app/Repositories/MessageRepository.php

```diff
@@ -120,6 +120,12 @@ public function updateMessagesForContest($contest, $messages)
                 $data['type'] = $type;
                 $data['key'] = $key;
 
+                if ($data['type'] === 'leaderboard') {
+                    if (! array_key_exists('show_images', $data)) {
+                        $data['show_images'] = 0;
```

Comment:
This is a weird hack i had to do because, for checkboxes, when you submit them unchecked the value doesn't get sent in the request. So what this does, is check if the `show_key` value was sent in the request, if so, then great use that value (1). If it wasn't sent, then that means the checkbox wasn't checked and we add the `show_images` key   to the request and set it to false.